editor/scene_tree_dock.cpp

```diff
@@ -2263,31 +2331,114 @@ void SceneTreeDock::_shader_creation_closed() {
+	if (selection.size() == 1) {
+		List<Node *>::Element *e = selection.front();
+		if (e) {
+			_toggle_editable_children(e->get());
+		}
+	} else {
+		for (Node *node : selection) {
+			if (node) {
+				bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+				editable = !editable;
+
+				if (editable == dialog_option_enable_all) {
+					_toggle_editable_children(node);
+				}
+			}
+		}
 	}
+	scene_tree->update_tree();
 }
 
-void SceneTreeDock::_toggle_placeholder_from_selection() {
+void SceneTreeDock::_disable_all_placeholder_from_selection() {
+	placeholder_enable_or_disable_dialog->hide();
+	dialog_option_enable_all = false;
+	_toggle_placeholder_from_selection();
+}
+
+void SceneTreeDock::_enable_all_placeholder_from_selection() {
+	dialog_option_enable_all = true;
+
 	List<Node *> selection = editor_selection->get_selected_node_list();
-	List<Node *>::Element *e = selection.front();
 
-	if (e) {
-		Node *node = e->get();
+	// Check if the selection contains nodes with editable children.
+	bool has_editables = false;
+	for (const Node *node : selection) {
 		if (node) {
-			_toggle_editable_children(node);
+			bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+			if (editable) {
+				has_editables = true;
+				break;
+			}
+		}
+	}
 
-			bool placeholder = node->get_scene_instance_load_placeholder();
-			placeholder = !placeholder;
+	if (has_editables) {
+		// Fire confirmation dialog.
+		placeholder_editable_instance_remove_dialog->set_text(TTR("Selection contains some nodes that have \"Editable Children\" enabled.\nEnabling \"Load as Placeholder\" will disable \"Editable Children\" and cause all properties of those nodes to be reverted to their default."));
+		placeholder_editable_instance_remove_dialog->popup_centered();
+	} else {
+		_toggle_placeholder_from_selection();
+	}
+}
 
-			node->set_scene_instance_load_placeholder(placeholder);
-			scene_tree->update_tree();
+void SceneTreeDock::_toggle_placeholder_from_selection() {
+	List<Node *> selection = editor_selection->get_selected_node_list();
+
+	if (selection.size() == 1) {
+		List<Node *>::Element *e = selection.front();
+		if (e) {
+			Node *node = e->get();
+			if (node) {
+				bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+				if (editable) {
+					_toggle_editable_children(node);
+
+					bool placeholder = node->get_scene_instance_load_placeholder();
+					placeholder = !placeholder;
+
+					node->set_scene_instance_load_placeholder(placeholder);
+				}
+			}
+		}
+	} else {
+		for (Node *node : selection) {
+			if (node) {
+				if (dialog_option_enable_all == true) {
+					bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+					if (editable) {
+						_toggle_editable_children(node);
+					}
+				}
+
+				bool placeholder = node->get_scene_instance_load_placeholder();
+				placeholder = !placeholder;
```

Comment:
```suggestion
				bool placeholder = !node->get_scene_instance_load_placeholder();
```

Reply:
Actually this can be simplified to
```C++
bool placeholder = node->get_scene_instance_load_placeholder();
if (placeholder != dialog_option_enable_all) {
```
I don't get why you are inverting bool and checking for equality.

Reply:
Because this line:
```
node->set_scene_instance_load_placeholder(placeholder);
```

That one would have to be inverted instead

Reply:
Then I think it should be `node->set_scene_instance_load_placeholder(dialog_option_enable_all)`?

Reply:
I think it might be more confusing than the current code, but that would work

Reply:
ok, I now have

```
bool placeholder = node->get_scene_instance_load_placeholder();

if (placeholder != dialog_option_enable_all) {
	node->set_scene_instance_load_placeholder(!placeholder);
}
```
the `if` is so as to not set placeholder to the same value it already is - I guess I thought it might avoid godot doing unnecessary work when the value didn't change. I think this reasoning was flawed, so probably good to just remove the `if`?

```
bool placeholder = node->get_scene_instance_load_placeholder();
node->set_scene_instance_load_placeholder(!placeholder);
```

Reply:
Well the `set_scene_instance_load_placeholder()` code is so minimal that the condition does matter indeed.

Reply:
realised that `if` was also making it work like a toggle. correcting my last comment, it can now just be one line:

```
node->set_scene_instance_load_placeholder(dialog_option_enable_all);
```